[2024-12-26T00:12:47.041+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2024-12-26T00:12:47.191+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: api_to_kafka_pipeline.fetch_and_produce_data scheduled__2024-12-25T23:00:00+00:00 [queued]>
[2024-12-26T00:12:47.212+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: api_to_kafka_pipeline.fetch_and_produce_data scheduled__2024-12-25T23:00:00+00:00 [queued]>
[2024-12-26T00:12:47.213+0000] {taskinstance.py:2866} INFO - Starting attempt 1 of 2
[2024-12-26T00:12:47.362+0000] {taskinstance.py:2889} INFO - Executing <Task(PythonOperator): fetch_and_produce_data> on 2024-12-25 23:00:00+00:00
[2024-12-26T00:12:47.450+0000] {logging_mixin.py:190} WARNING - /home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py:70 DeprecationWarning: This process (pid=6236) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2024-12-26T00:12:47.457+0000] {standard_task_runner.py:72} INFO - Started process 6244 to run task
[2024-12-26T00:12:47.450+0000] {standard_task_runner.py:104} INFO - Running: ['airflow', 'tasks', 'run', 'api_to_kafka_pipeline', 'fetch_and_produce_data', 'scheduled__2024-12-25T23:00:00+00:00', '--job-id', '61', '--raw', '--subdir', 'DAGS_FOLDER/api_to_kafka_dag.py', '--cfg-path', '/tmp/tmp4qehsi9a']
[2024-12-26T00:12:47.460+0000] {standard_task_runner.py:105} INFO - Job 61: Subtask fetch_and_produce_data
[2024-12-26T00:12:48.212+0000] {task_command.py:467} INFO - Running <TaskInstance: api_to_kafka_pipeline.fetch_and_produce_data scheduled__2024-12-25T23:00:00+00:00 [running]> on host 0f607e57b1dd
[2024-12-26T00:12:48.389+0000] {taskinstance.py:3132} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='airflow' AIRFLOW_CTX_DAG_ID='api_to_kafka_pipeline' AIRFLOW_CTX_TASK_ID='fetch_and_produce_data' AIRFLOW_CTX_EXECUTION_DATE='2024-12-25T23:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-12-25T23:00:00+00:00'
[2024-12-26T00:12:48.391+0000] {taskinstance.py:731} INFO - ::endgroup::
[2024-12-26T00:12:48.707+0000] {logging_mixin.py:190} INFO - âœ… Data fetched successfully for IBM at 5min interval.
[2024-12-26T00:12:48.708+0000] {logging_mixin.py:190} INFO - ðŸš€ Sending data to Kafka...
[2024-12-26T05:36:42.048+0000] {job.py:229} INFO - Heartbeat recovered after 19428.15 seconds
[2024-12-26T05:37:07.139+0000] {local_task_job_runner.py:346} WARNING - State of this instance has been externally set to up_for_retry. Terminating instance.
[2024-12-26T05:37:07.141+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
[2024-12-26T05:37:07.146+0000] {process_utils.py:132} INFO - Sending 15 to group 6244. PIDs of all processes in the group: [6244]
[2024-12-26T05:37:07.150+0000] {process_utils.py:87} INFO - Sending the signal 15 to group 6244
[2024-12-26T05:38:07.180+0000] {process_utils.py:150} WARNING - process psutil.Process(pid=6244, name='airflow task ru', status='sleeping', started='00:12:46') did not respond to SIGTERM. Trying SIGKILL
[2024-12-26T05:38:07.186+0000] {process_utils.py:87} INFO - Sending the signal 9 to group 6244
[2024-12-26T05:38:07.194+0000] {process_utils.py:80} INFO - Process psutil.Process(pid=6244, name='airflow task ru', status='terminated', exitcode=<Negsignal.SIGKILL: -9>, started='00:12:46') (6244) terminated with exit code -9
[2024-12-26T05:38:07.197+0000] {standard_task_runner.py:190} ERROR - ('Job 61 was killed before it finished (likely due to running out of memory)', 'For more information, see https://airflow.apache.org/docs/apache-airflow/stable/troubleshooting.html#LocalTaskJob-killed')
